### PR TITLE
Updated Cicero template, now using MonetaryValue instead of Double

### PR DESCRIPTION
**Pull request by 4katie**
Signed-off-by: Katie <sa951009@gmail.com>

# Issue #254 
Updated the installment-sale Cicero template to use MonetaryValue (doubleValue, currencyCode) instead of just double for all of the monetary amounts. 

### Changes
- all variables now have type MonetaryValue instead of Double
- changed template text, model, and logic to reflect this change
- tests have also been changed to reflect the changes 

### Flags
- N/A

### Related Issues
Cicero issue [#423](https://github.com/accordproject/cicero/issues/423#issue-500945851): Template Tests are not able to override the default timeout

### CODE REVIEW

File: src/installment-sale/lib/logic.ergo [hunk 1]
```diff
@@ -17,70 +17,81 @@ namespace org.accordproject.installmentsale
...
+    let total_paid = MonetaryAmount{ doubleValue : state.total_paid.doubleValue + request.amount.doubleValue, currencyCode : contract.INITIAL_DUE.currencyCode};
+
     set state InstallmentSaleState{
       stateId: "#1",
       status: "Fulfilled",
       balance_remaining: balance,
       total_paid: total_paid,
       next_payment_month: 0
     };
-    emit PaymentObligation{
```
[the remark this answers is not in the document]
Comment (4katie) [reply src/installment-sale/lib/logic.ergo]: So `PaymentObligation` for the `PayLastInstallment` clause was deleted, because I thought it doesn't make sense to have a payment obligation after making the 'last installment', especially since the description states:
`contract.BUYER.partyId ++ " should pay installment to " ++ contract.SELLER.partyId` when there is no installment left.

If it is beneficial to emit a message nonetheless, maybe we should change the message to 'CompletedPaymentObligation' or something like that? :O

File: src/installment-sale/lib/logic.ergo [hunk 2]
```diff
@@ -17,70 +17,81 @@ namespace org.accordproject.installmentsale
 import org.accordproject.cicero.runtime.*
 import org.accordproject.money.MonetaryAmount
 
+define function round(x:Double) {
```
[the remark this answers is not in the document]
Comment (4katie) [reply src/installment-sale/lib/logic.ergo]: Hmm, how should that be done? 
For the last installment, it is easy to create an `enforce` statement because we know exactly how much it should be done. 
For the other payments, should it be done by using the `roundn` function for all of the values, like before_interest, balance, and so on?

File: src/installment-sale/lib/logic.ergo [hunk 3]
```diff
@@ -40,13 +40,15 @@ contract InstallmentSale over InstallmentSaleContract state InstallmentSaleState
     enforce (contract.INITIAL_DUE.currencyCode = contract.TOTAL_DUE_BEFORE_CLOSING.currencyCode) 
         and (contract.INITIAL_DUE.currencyCode = contract.MIN_PAYMENT.currencyCode) 
         and (contract.INITIAL_DUE.currencyCode = contract.DUE_AT_CLOSING.currencyCode);
```
Comment (jeromesimeon) [inline src/installment-sale/lib/logic.ergo]: The syntax is either:
```
enforce Expr;
StmtWhenSuccess
```
or
```
enforce Expr else StmtWhenFailure;
StmtWhenSuccess
````

so this version has an extraneous `;`

Comment (jeromesimeon) [reply src/installment-sale/lib/logic.ergo]: I believe this is the reason tests in Travis now fail.

File: src/installment-sale/lib/logic.ergo (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeromesimeon) [reply src/installment-sale/lib/logic.ergo]: I was thinking it could be done using the same `roundn` function used here, but it's not mandatory. Just wondered why do it in one place, but not the rest of the code?

File: src/installment-sale/lib/logic.ergo (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeromesimeon) [reply src/installment-sale/lib/logic.ergo]: > So `PaymentObligation` for the `PayLastInstallment` clause was deleted, because I thought it doesn't make sense to have a payment obligation after making the 'last installment', especially since the description states:
> `contract.BUYER.partyId ++ " should pay installment to " ++ contract.SELLER.partyId` when there is no installment left.
> 
> If it is beneficial to emit a message nonetheless, maybe we should change the message to 'CompletedPaymentObligation' or something like that? :O

That's interesting! I had not thought of it that way. I think this is something that we might want to explain in the accord project documentation in a better way, but the payment obligation is something that is emitted so an external service could act on the actual payment.

In other words, the request for payment is checked by the contract and emits a real-world payment obligation (which can then be handed over to an online payment system, or so someone could write a check).

So concretely, it means the payment request occurs _before_ the actual payment occurs, which may be confusing in this particular template, but is consistent with the current programming model.

I hope that makes sense.

**Comment by jeromesimeon** [comment]
Congrats on your first template @4katie ! wd
